src/components/slider/draw.js

```diff
@@ -0,0 +1,507 @@
+}
+
+function drawLabel(item, data, sliderOpts) {
+    var text = item.selectAll('text')
+        .data([0]);
+
+    text.enter().append('text')
+        .classed(constants.labelClass, true)
+        .classed('user-select-none', true)
+        .attr('text-anchor', 'middle');
+
+    text.call(Drawing.font, sliderOpts.font)
+        .text(data.step.label)
+        .call(svgTextUtils.convertToTspans);
+
+    return text;
+}
+
+function drawLabelGroup(sliderGroup, sliderOpts) {
+    var labels = sliderGroup.selectAll('g.' + constants.labelsClass)
+        .data([0]);
+
+    labels.enter().append('g')
+        .classed(constants.labelsClass, true);
+
+    var labelItems = labels.selectAll('g.' + constants.labelGroupClass)
+        .data(sliderOpts.labelSteps);
+
+    labelItems.enter().append('g')
+        .classed(constants.labelGroupClass, true);
+
+    labelItems.exit().remove();
+
+    labelItems.each(function(d) {
+        var item = d3.select(this);
+
+        item.call(drawLabel, d, sliderOpts);
+
+        Lib.setTranslate(item,
+            normalizedValueToPosition(sliderOpts, d.fraction),
+            constants.tickOffset + constants.tickLength + sliderOpts.labelHeight
+        );
+    });
+
+}
+
+function handleInput(gd, sliderGroup, sliderOpts, normalizedPosition, doTransition) {
+    var quantizedPosition = Math.round(normalizedPosition * (sliderOpts.steps.length - 1));
+
+
+    if(quantizedPosition !== sliderOpts.active) {
+        setActive(gd, sliderGroup, sliderOpts, quantizedPosition, true, doTransition);
+    }
+}
+
+function setActiveByLabel(gd, sliderGroup, sliderOpts, label, doCallback, doTransition) {
+    var index;
+    for(var i = 0; i < sliderOpts.steps.length; i++) {
+        var step = sliderOpts.steps[i];
+        if(step.label === label) {
+            index = i;
+            break;
+        }
+    }
+
+    if(index !== undefined) {
+        setActive(gd, sliderGroup, sliderOpts, index, doCallback, doTransition);
+    }
+}
+
+function setActive(gd, sliderGroup, sliderOpts, index, doCallback, doTransition) {
+    sliderOpts._input.active = sliderOpts.active = index;
+
+    sliderGroup.call(setGripPosition, sliderOpts, sliderOpts.active / (sliderOpts.steps.length - 1), doTransition);
+
+    var step = sliderOpts.steps[sliderOpts.active];
+
+    if(step && step.method && doCallback) {
+        var args = step.args;
+        Plotly[step.method](gd, args[0], args[1], args[2]).catch(function() {
+            // This is not a disaster. Some methods like `animate` reject if interrupted
+            // and *should* nicely log a warning.
+            Lib.warn('Warning: Plotly.' + step.method + ' was called and rejected.');
+        });
+    }
+}
+
+function attachGripEvents(item, gd, sliderGroup, sliderOpts) {
+    var node = sliderGroup.node();
+    var $gd = d3.select(gd);
+
+    item.on('mousedown', function() {
+        var grip = sliderGroup.select('.' + constants.gripRectClass);
+
+        d3.event.stopPropagation();
```

Comment:
we might want to use a `coverSlip` in the range slider part II [here](https://github.com/plotly/plotly.js/pull/962/files#r79949182) to avoid selecting other elements on the graph on drag.